### ISSUE TITLE
Puppet references non-root user, but does not include one in image

### ISSUE THREAD

**Issue by Chuxel** [microsoft/vscode-dev-containers#334]
@jpogran I noticed an issue in a regression test. Currently the puppet definition uses the `puppet/pdk:latest` image. However, `devcontainer.json` includes a reference to a `vscode` user that does not exist in the image.

```json
	// Uncomment to connect as a non-root user. See https://aka.ms/vscode-remote/containers/non-root.
	// "remoteUser": "vscode"
```

I'd suggest either removing this comment or adding the non-root user in the Dockerfile. You could either follow instructions [here](https://aka.ms/vscode-remote/containers/non-root) or reference the common script. There's options on it to determine disable installing certain things if you so choose. 

For example, this would only add the non-root user
```Dockerfile
ARG COMMON_SCRIPT_SOURCE="https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/script-library/common-debian.sh"
ARG COMMON_SCRIPT_SHA="dev-mode"

RUN apt-get update \
    && export DEBIAN_FRONTEND=noninteractive \
    && apt-get -y install --no-install-recommends curl ca-certificates 2>&1 \
    && curl -sSL  ${COMMON_SCRIPT_SOURCE} -o /tmp/common-setup.sh \
    && ([ "${COMMON_SCRIPT_SHA}" = "dev-mode" ] || (echo "${COMMON_SCRIPT_SHA} /tmp/common-setup.sh" | sha256sum -c -)) \
    && /bin/bash /tmp/common-setup.sh "false" "vscode" "1000" "1000" "false" \
    && rm /tmp/common-setup.sh
```

More info [here](https://github.com/microsoft/vscode-dev-containers/blob/master/script-library).  

Thanks for the contribution here!

**Comment by jpogran** [comment]
Thanks for catching this, I hadn't noticed that got added after the initial commit.

I think I have to remove the reference in the `devcontainer.json`, as I don't control the docker image creation part. I will make a ticket with that team to see if this can be added to the build.

**Comment by jpogran** [comment]
For reference, this is the [dockerfile](https://github.com/puppetlabs/pdk-docker/blob/master/Dockerfile)

**Comment by Chuxel** [comment]
Resolved by #335!  Thanks!
